### PR TITLE
feat: add ability to select migration files when committing

### PR DESCRIPTION
**Pull request by AlexanderYarushin**
Syntax:
8base migration commit --target file1.tsc --target file2.tsc --target file3.tsc

### CODE REVIEW

**Comment by Selnapenek** [comment]
@AlexanderYarushin can you please add a description to the man/help

File: src/engine/commands/migration/commands/commit.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import * as yargs from 'yargs';
 import { CommitMode, MigrateMode, RequestOptions } from '../../../../interfaces/Common';
 import { DEFAULT_ENVIRONMENT_NAME } from '../../../../consts/Environment';
 import { Interactive } from '../../../../common/interactive';
+import * as fs from 'fs';
+const path = require('path');
```
Comment (Selnapenek) [inline src/engine/commands/migration/commands/commit.ts:12]: what is the problem with the import path?

File: src/engine/commands/migration/commands/commit.ts [hunk 2]
```diff
@@ -40,10 +43,26 @@ export default {
         ? await uploadProject(context, options)
         : { buildName: null };
 
+      let migrationNames: string[] = [];
```
Comment (Selnapenek) [inline src/engine/commands/migration/commands/commit.ts:46]: this functionality should works only for modes `CommitMode.FULL and  ONLY_MIGRATIONS'

File: src/locales/en.ts [hunk 3]
```diff
@@ -397,7 +397,8 @@ export default {
     migration_commit_mode_describe: 'Commit mode.',
     migration_commit_dest_env_master: 'Environment you want to commit is Master. Are you sure you want to continue?',
     migration_commit_canceled: 'Commit canceled',
-
+    migration_commit_file_does_not_exist: 'Migrations file with name "{{- name}}" doesn\'t exist',
```
Comment (Selnapenek) [inline src/locales/en.ts:400]: `{{- name}}` for what we need `-`?

File: src/engine/commands/migration/commands/commit.ts [hunk 4]
```diff
@@ -40,10 +43,28 @@ export default {
         ? await uploadProject(context, options)
         : { buildName: null };
 
+    if (params.mode === CommitMode.ONLY_PROJECT || params.mode === CommitMode.FULL) {
+      var migrationNames: string[] = [];
```
Comment (Selnapenek) [inline src/engine/commands/migration/commands/commit.ts:47]:
```suggestion
   let migrationNames: string[] | undefined = undefined; 
    if (params.mode === CommitMode.ONLY_PROJECT || params.mode === CommitMode.FULL) {
      migrationNames = [];
```

Comment (Selnapenek) [reply src/engine/commands/migration/commands/commit.ts:47]: please don't use var

Comment (AlexanderYarushin) [reply src/engine/commands/migration/commands/commit.ts:47]: Because of the global scope ? Or are there other reasons

Comment (Selnapenek) [reply src/engine/commands/migration/commands/commit.ts]: yes

File: src/engine/commands/migration/commands/commit.ts [hunk 5]
```diff
@@ -40,10 +43,28 @@ export default {
...
         : { buildName: null };
 
+    let migrationNames: string[] | undefined = undefined;
+    if (params.mode === CommitMode.ONLY_PROJECT || params.mode === CommitMode.FULL) {
+      migrationNames = [];
+      if (typeof params.target === 'string') {
+        migrationNames.push(params.target);
+      } else if (typeof params.target === 'object') {
+        migrationNames = params.target;
+      }
```
Comment (Selnapenek) [inline src/engine/commands/migration/commands/commit.ts:53]: looks strange, try to define option `target` as array

**Comment by Selnapenek** [comment]
please do not forget to update the e2e (`help.test.ts`)

**Comment by AlexanderYarushin** [comment]
I didn't understand about `help.test.ts`. It checks the output of --help ? But then I need to change `help.test.ts.snap` so that `migration commit --help` corresponds to the output.

File: src/engine/commands/migration/commands/commit.ts [hunk 6]
```diff
@@ -8,6 +8,9 @@ import * as yargs from 'yargs';
 import { CommitMode, MigrateMode, RequestOptions } from '../../../../interfaces/Common';
 import { DEFAULT_ENVIRONMENT_NAME } from '../../../../consts/Environment';
 import { Interactive } from '../../../../common/interactive';
+import * as fs from 'fs';
```
Comment (s1owjke) [inline src/engine/commands/migration/commands/commit.ts:11]: Order of imports

File: src/engine/commands/migration/commands/commit.ts [hunk 7]
```diff
@@ -30,6 +33,22 @@ export default {
       }
     }
 
+    if (params.mode === CommitMode.ONLY_PROJECT && params.target) {
```
Comment (s1owjke) [inline src/engine/commands/migration/commands/commit.ts:36]: Same for ONLY_PLUGINS mode

File: src/engine/commands/migration/commands/commit.ts [hunk 8]
```diff
@@ -30,6 +33,22 @@ export default {
...
+      throw new Error(context.i18n.t('migration_commit_in_project_mode'));
+    }
+
+    const migrationNames: string[] | undefined = params.target;
+
+    if (migrationNames) {
+      const paths: PredefineData = new PredefineData();
```
Comment (eugeneantonevich) [inline src/engine/commands/migration/commands/commit.ts:43]: StaticConfig.rootExecutionDir

**Review by s1owjke — APPROVED** [review]

**Pull request merged**